### PR TITLE
predict: add --resume so an interrupted whole-scroll run is not lost

### PR DESCRIPTION
**Pull request by TAUIL-Abd-Elilah**
## The problem

A whole-scroll inference pass streams for hours. Measuring the I/O side of it in #1325: PHerc1218 is 1.34 TB, and at the 11 MB/s my connection sustains that is ~36 h even under the best traversal policy, ~43 days with no chunk cache at all. At that length interruption is ordinary — a dropped connection, a reboot, a laptop lid.

Today an interrupted run is worth nothing. `_create_output_stores` opens `logits_part_{id}.zarr` with `mode='w'`, which recreates the array, so restarting the same command discards every patch already computed.

I measured it rather than assuming it. A 384³ region of PHerc1218 at level 0, 216 patches: killed the run after 182 s with 64 patches on disk, then reran the identical command and sampled the store every 5 s.

```
t=0s  64 chunks     t=20s   0 chunks   <- store recreated
t=5s  64 chunks     t=30s   3 chunks
t=10s 64 chunks     t=35s   5 chunks
t=15s 64 chunks     t=40s   7 chunks
```

The count holds, collapses to zero, and starts again from three. `--num_parts`/`--part_id` do not help: they split the work up front, and a dead shard is still a dead shard.

## What this adds

`--resume` continues a previous run in the same `--output_dir`: patches already written are skipped, and the existing logits store is appended to rather than recreated.

```
vesuvius.predict --input_dir <url> --output_dir out --bbox z0:z1,y0:y1,x0:x1 --resume
```

Two details drove the design and both are load-bearing.

**Completion is recorded in a manifest, not inferred by listing the store.** The logits array is chunked one-chunk-per-patch, so it is tempting to treat "chunk exists" as "patch done". That is wrong here: `write_empty_chunks=False` is set deliberately, so a patch that produces all zeros writes no chunk at all. The store therefore cannot distinguish *done and empty* from *not done* — and on a masked scroll volume the empty patches are the majority, so a listing-based resume would rerun all of them on every restart, forever. Empty patches are submitted to the writer like any other, so `on_written` records them correctly. `completed_part_{id}.json` sits beside the store and is rewritten atomically.

**A resume whose settings differ is refused, not merged.** The signature covers the input volume, bbox, patch size, overlap, model, TTA setting, normalization, part split and patch count, and the error names the fields that differ:

```
--resume refused: out/completed_part_0.json was written by a different run.
  differing: overlap, tta
```

One store holding patches computed two different ways, with nothing recording the split, is the failure mode behind #1250 and #1253 — a published prediction that turned out to reproduce only under a TTA setting no artifact mentioned. It seemed worth making that unrepresentable rather than documented.

The manifest is flushed on whichever comes first, 16 writes or 30 seconds, rather than per patch — so a crash costs a bounded handful of recomputed patches instead of an fsync on every write, and the time limit bounds the loss regardless of throughput.

`BoundedPatchWriter` gains an optional `on_written(index)` callback, fired only after the store accepts the patch and never from the error path — a resume that trusted a patch the store rejected would leave a permanent hole no later run goes back for. It is kept separate from `on_progress`, which is a bare counter for the progress bar and has no index to give.

Patch indices, the coordinates store and its attributes are unchanged, so `blend_logits` and `finalize_outputs` need no flags and see exactly what they see today.

## Tests

`vesuvius/tests/models/run/test_inference_resume.py`, 20 cases: manifest round trip, missing and corrupt manifests, refusal on each differing field, per-part manifests, that resume reopens the store rather than recreating it, that a plain run still recreates it, that a store of the wrong shape is refused, that a patch which wrote no chunk is still recorded complete, that a manifest built from numpy scalars serialises, that no `.tmp` survives a failed write, and a guard that parses `__init__` for `self.X =` and asserts the run signature reads nothing else.

Nearby suites pass unchanged: `tests/models/run` is 67 passed, and `tests/models/run` plus `tests/data` together are 119 passed, 1 skipped.

## End to end

Same region, same interruption, with the flag. Run 1 was killed with 54 patch chunks on disk and a manifest recording 51 complete; run 2 was the identical command with `--resume`:

```
--resume: 51 of 216 patches already written; skipping them
```

Chunks on disk through the restart: `54, 54, 54, 54, 54, 54, 54, 54, 54, 55, 57, 61, 62, 64 …`

It holds flat while the resumed run works through what is left, then climbs — against `64, 64, 64, 0, 3, 5, 7` for the same sequence without the flag.

Three things this caught that the unit tests did not, all fixed here:

- `patch_size` and `num_classes` arrive as numpy integers, and `json.dump` raises on those *after* opening the file — leaving a zero-byte `.tmp`, no manifest, and a run that looked fine.
- Flushing the manifest every N writes is not enough on its own: the first run was interrupted after 63 writes with the interval at 64, so nothing was ever persisted. It now flushes on 16 writes **or** 30 seconds, whichever comes first, and on the way out of Ctrl-C or an exception.
- The signature read `self.input_dir` and `self.disable_tta`, neither of which exists — the instance stores `self.input` and `self.do_tta`. There is now a test that parses `__init__` for `self.X =` and asserts the signature reads nothing else; it caught the second one immediately after the first was fixed by hand.

Related: #1241 and #1247 added `--bbox` and made an ROI run select from the full-volume grid; #1253 recorded TTA provenance in the output attrs. This is the same thread — making a long streamed run something you can trust and finish.

### CODE REVIEW

**Comment by bruniss** [comment]
most of our inference uses s3 paths which this does not handle, and there are many areas where the resume functionality as written could easily overwrite the exact stores it's trying to resume from.
